extensions/amp-story/1.0/media-tasks.js

```diff
@@ -109,200 +108,132 @@ function copyAttributes(fromEl, toEl) {
 }
 
 /**
- * Base class for tasks executed in order on HTMLMediaElements.
+ * A task executed in order on HTMLMediaElements.
+ * @interface
  */
 export class MediaTask {
-  /**
-   * @param {string} name
-   * @param {!Object=} options
-   */
-  constructor(name, options = {}) {
-    /** @private @const {string} */
-    this.name_ = name;
-
-    const deferred = new Deferred();
-
-    /** @private @const {!Promise} */
-    this.completionPromise_ = deferred.promise;
-
-    /** @protected @const {!Object} */
-    this.options = options;
-
-    /** @private {?function()} */
-    this.resolve_ = deferred.resolve;
-
-    /** @private {?function(*)} */
-    this.reject_ = deferred.reject;
-  }
-
-  /**
-   * @return {string} The name of this task.
-   */
-  getName() {
-    return this.name_;
-  }
-
-  /**
-   * @return {!Promise<*>} A promise that is resolved when the task has
-   *     completed execution.
-   */
-  whenComplete() {
-    return this.completionPromise_;
-  }
-
-  /**
-   * @param {!HTMLMediaElement} mediaEl The element on which this task should be
-   *     executed.
-   * @return {!Promise} A promise that is resolved when the task has completed
-   *     execution.
-   */
-  execute(mediaEl) {
```

Comment:
`execute()` is merely `Promise.resolve`, which is now handled on `executeNextMediaTask_()`.